From 4840925220ef618d81aded255016c9b5041e09b9 Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Fri, 31 Mar 2023 14:56:49 +0100
Subject: [PATCH] dtoverlays: Add link-frequency parameter for the Sony IMX708
 sensor

Add a parameter to change the sensor device CSI-2 link frequency to
one of the following values: 450Mhz (default), 447Mhz, or 453Mhz.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README             | 2 ++
 arch/arm/boot/dts/overlays/imx708-overlay.dts | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 50f14e009d2f..875b58ab1723 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2417,6 +2417,8 @@ Params: rotation                Mounting rotation of the camera sensor (0 or
                                 configuring the sensor (default on)
         cam0                    Adopt the default configuration for CAM0 on a
                                 Compute Module (CSI0, i2c_vc, and cam0_reg).
+        link-frequency          Allowable link frequency values to use in Hz:
+                                450000000 (default), 447000000, 453000000.
 
 
 Name:   iqaudio-codec
diff --git a/arch/arm/boot/dts/overlays/imx708-overlay.dts b/arch/arm/boot/dts/overlays/imx708-overlay.dts
index 9107e985c2a8..6efbe0943211 100644
--- a/arch/arm/boot/dts/overlays/imx708-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx708-overlay.dts
@@ -88,6 +88,7 @@ __overrides__ {
 		       <&vcm_node>, "VDD-supply:0=",<&cam0_reg>;
 		vcm = <&vcm_node>, "status",
 		      <0>, "=4";
+		link-frequency = <&cam_endpoint>,"link-frequencies#0";
 	};
 };
 
-- 
2.35.5

